pkg/daemon/ceph/client/osd.go

```diff
@@ -220,6 +221,45 @@ func GetOSDUsage(context *clusterd.Context, clusterInfo *ClusterInfo) (*OSDUsage
 	return &osdUsage, nil
 }
 
+func convertKibibytesToTebibytes(kib string) (float64, error) {
+	kibFloat, err := strconv.ParseFloat(kib, 64)
+	if err != nil {
+		return float64(0), errors.Wrap(err, "failed to convert string to float")
+	}
+	return kibFloat / float64(1024*1024*1024), nil
+}
+
+func ResizeOsdCrushWeight(actualOSD OSDNodeUsage, ctx *clusterd.Context, clusterInfo *ClusterInfo) (bool, error) {
+	crushWeight, err := strconv.ParseFloat(actualOSD.CrushWeight.String(), 64)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed converting string to float for osd.%d crush weight %q", actualOSD.ID, actualOSD.CrushWeight.String())
+	}
+	// actualOSD.KB is in KiB units
+	desiredOSDTiB, err := convertKibibytesToTebibytes(actualOSD.KB.String())
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to convert KiB to TiB for osd.%d crush weight %q", actualOSD.ID, actualOSD.KB.String())
+	}
+
+	// do not reweight if the desired crush weight is 0 or they are equal or there percentage resize is less than 10 percent
+	if desiredOSDTiB == float64(0) {
+		logger.Debugf("desired osd size is 0 for osd.%d, not resizing the crush weights", actualOSD.ID)
```

Comment:
This isn't really the same as the "desired" size by the user right? It's the value that is reported by `ceph osd df`? Can we change the wording from "desired osd size" to something that indicates that this is the value reported by Ceph?

Is this a condition you've seen in testing? It seems like an error condition to me. I would think that this should be an Info or possibly even Warning log. But returning an error could be appropriate. I'm not sure what Ceph conditions could cause this IRL, but I'm guessing the only possible option would be some kind of degraded OSD state, and Rook might have to keep trying this until the degraded state is resolved.

@travisn other thoughts?

Reply:
IIRC Parth saw the 0 size during the first reconcile, basically when the OSD is being created, so we just want to skip any updates when it is 0. But anyway, if we add a check to skip the updates when the new size is less than the current size, seems like we won't need to check specifically for 0.

Reply:
> This isn't really the same as the "desired" size by the user right? It's the value that is reported by ceph osd df? Can we change the wording from "desired osd size" to something that indicates that this is the value reported by Ceph?

I think this got missed with a focus on just the 2nd part

Reply:
What I see in the unit test it does work, if want I can explicitly check